src/storage/src/source/mysql/replication.rs

```diff
@@ -539,20 +533,8 @@ async fn raw_stream<'a>(
     ))
     .await?;
 
-    // Generate a deterministic server-id for identifying us as a replica on the upstream mysql server.
-    // The value does not actually matter since it's irrelevant for GTID-based replication and won't
-    // cause errors if it happens to be the same as another replica in the mysql cluster (based on testing),
-    // but by setting it to a constant value we can make it easier for users to identify Materialize connections
-    let server_id = match config.id {
-        GlobalId::System(id) => id,
-        GlobalId::User(id) => id,
-        GlobalId::Transient(id) => id,
-        _ => unreachable!(),
-    };
-    let server_id = match u32::try_from(server_id) {
-        Ok(id) if id + REPLICATION_SERVER_ID_OFFSET < u32::MAX => id + REPLICATION_SERVER_ID_OFFSET,
-        _ => REPLICATION_SERVER_ID_OFFSET,
-    };
+    let mut rng = rand::thread_rng();
+    let server_id: u32 = rng.gen();
```

Comment:
The comment above says:

> The value does not actually matter since it's irrelevant for GTID-based replication and won't
> cause errors if it happens to be the same as another replica in the mysql cluster

Based on this I'd think it should be fine if all replicas running a source used the same server ID. Is that not the case?

Reply:
I thought I had reverted this change. It does matter in the end, in the sense that if two clients use the same server_id at the same time one of them will get an error. Choosing a random server id was my initial remedy for this issue, but afterwards I thought it was hacky enough that I removed MySQL from being replicated in this PR and I will treat it like postgres, i.e a source that gets scheduled only on the most recent replica of a cluster